File: project/jimmer-client/src/test/java/org/babyfish/jimmer/client/model/Author.java
package org.babyfish.jimmer.client.model;

import org.babyfish.jimmer.sql.Entity;
import org.babyfish.jimmer.sql.Id;
import org.babyfish.jimmer.sql.ManyToMany;

import java.util.List;

@Entity
public interface Author {

    @Id
    long id();

    String firstName();

    String lastName();

    Gender gender();

    @ManyToMany(mappedBy = "authors")
    List<Book> books();
}

File: project/jimmer-sql-kotlin/src/test/kotlin/org/babyfish/jimmer/sql/kt/model/Author.kt
package org.babyfish.jimmer.sql.kt.model

import org.babyfish.jimmer.sql.*
import javax.validation.constraints.NotBlank

@Entity
interface Author {

    @Id
    val id: Long

    @Key
    val firstName: @NotBlank String

    @Key
    val lastName: @NotBlank String

    val fullName: String
        get() = "$firstName $lastName"

    val gender: Gender

    @ManyToMany(mappedBy = "authors")
    val books: List<Book>

    @Transient
    val organization: Organization

    fun unusedFun() {}
}